torch/tensor.py

```diff
@@ -28,6 +28,11 @@ def type_as(self, tensor):
         Params:
             tensor (Tensor): the tensor which has the desired type
         """
+        from torch.autograd import Variable
+        if isinstance(tensor, Variable):
+            tensor = tensor.data
+        elif not torch.is_tensor(tensor):
+            raise ValueError('Expected a tensor in type_as, got type {}'.format(type(tensor)))
```

Comment:
Given how expensive `import`, `isinstance` and `torch.is_tensor` is, I'm not sure if we really want to include this patch. Maybe let's just wait until me unify Tensors and Variables?

Reply:
Sounds good!